mods/ra/maps/allies-06a/allies06a-AI.lua

```diff
@@ -0,0 +1,226 @@
+	hard = 3
+}
+
+WaterAttackTypes =
+{
+	easy = { "ss" },
+	normal = { "ss", "ss" },
+	hard = { "ss", "ss", "ss" }
+}
+
+VehicleTypes = { "v2rl", "3tnk", "3tnk", "3tnk", "3tnk", "harv" }
+
+InfTypes =
+{
+	{ "e1", "e1", "e1"},
+	{ "e2", "e1", "e1"},
+	{ "e4", "e4", "e1"}
+}
+
+AttackRallyPoints =
+{
+	{ SovietOreAttackStart.Location, SovietOreAttack1.Location },
+	{ SovietBaseAttack.Location },
+	{ SovietOreAttack2.Location }
+}
+
+ImportantBuildings = { WeaponsFactory, Airfield, dome2, SovietConyard }
+SovietAircraftType = { "yak" }
+Yaks = { }
+IdlingUnits = { }
+IdlingTanks = { tank1, tank2, tank3, tank4, tank5, tank6, tank7, tank8 }
+IdlingNavalUnits = { }
+
+InitialiseAttack = function()
+	Utils.Do(ImportantBuildings, function(a)
+		Trigger.OnDamaged(a, function()
+			Utils.Do(IdlingTanks, function(unit)
+				if not unit.IsDead then
+					unit.Hunt()
+				end
+			end)
+		end)
+	end)
+end
+
+Attack = 0
+ProduceInfantry = function()
+	if SovietBarracks.IsDead then
+		return
+	end
+
+	Attack = Attack + 1
+	local toBuild = Utils.Random(InfTypes)
+	ussr.Build(toBuild, function(units)
+		if Attack == 2 and not AttackTnk1.IsDead then
+			units[#units + 1] = AttackTnk1
+		elseif Attack == 4 and not AttackTnk2.IsDead then
+			units[#units + 1] = AttackTnk2
+		end
+
+		SendAttack(units, Utils.Random(AttackRallyPoints))
+		Trigger.AfterDelay(DateTime.Seconds(BuildDelays), ProduceInfantry)
+	end)
+end
+
+ProduceVehicles = function()
+	if WeaponsFactory.IsDead then
+		return
+	end
+	ussr.Build(VehicleTypes, function(units)
+		Utils.Do(units, function(unit)
+			if unit.Type ~= "harv" then
+				IdlingTanks[#IdlingTanks + 1] = unit
+			end
+		end)
+	end)
+end
+
+ProduceNaval = function()
+	if SubPen.IsDead then
+		return
+	end
+
+	ussr.Build(WaterAttackTypes, function(units)
+		Utils.Do(units, function(unit)
+			IdlingNavalUnits[#IdlingNavalUnits + 1] = unit
+		end)
+
+		Trigger.AfterDelay(DateTime.Minutes(1) + DateTime.Seconds(40), ProduceNaval)
+		if #IdlingNavalUnits >= WaterAttacks then
+			Trigger.AfterDelay(DateTime.Seconds(20), function()
+				SendAttack(SetupNavalAttackGroup(), { Harbor.Location })
+			end)
+		end
+	end)
+end
+
+ProduceAircraft = function()
+	if Airfield.IsDead then
+		return
+	end
+
+	ussr.Build(SovietAircraftType, function(units)
+		local yak = units[1]
+		Yaks[#Yaks + 1] = yak
+
+		Trigger.OnKilled(yak, ProduceAircraft)
+		if #Yaks == 1 then
+			Trigger.AfterDelay(DateTime.Seconds(BuildDelays), ProduceAircraft)
+		end
+
+		TargetAndAttack(yak)
+	end)
+end
+
+TargetAndAttack = function(yak, target)
+	if not target or target.IsDead or (not target.IsInWorld) then
+		local enemies = Utils.Where(Map.ActorsInWorld, function(self) return self.Owner == player and self.HasProperty("Health") end)
+		if #enemies > 0 then
+			target = Utils.Random(enemies)
+		else
+			yak.Wait(DateTime.Seconds(5))
+		end
+	end
+
+	if target and yak.AmmoCount() > 0 then
+		yak.Attack(target)
+	else
+		yak.ReturnToBase()
+	end
+
+	yak.CallFunc(function()
```

Comment:
Is that working properly even? From my tests, CallFunc breaks the activity queue and the sole reason this is unnoticable in say, ProductionAirdrop is because the actors affected there also has FlyAwayOnIdle.